### PR TITLE
Ruby 3 Compatibility

### LINKED ISSUE #2: bugs running with ruby 3
```
      ArgumentError:
       tried to create Proc object without a block
     # ./lib/cron_swanson/whenever.rb:86:in `new'
     # ./lib/cron_swanson/whenever.rb:86:in `schedule'
     # <eval>:2:in `initialize'
     # ./spec/lib/cron_swanson/whenever_spec.rb:177:in `new'
     # ./spec/lib/cron_swanson/whenever_spec.rb:177:in `block (4 levels) in <top (required)>'
     # ./spec/lib/cron_swanson/whenever_spec.rb:170:in `each'
     # ./spec/lib/cron_swanson/whenever_spec.rb:170:in `block (3 levels) in <top (required)>'
```

### PR DESCRIPTION
**Pull request by mzagaja**
Resolve alexdean/cron_swanson#2 by migrating &Proc.new to &block per advice at https://blog.saeloun.com/2019/09/02/ruby-2-7-proc-without-block-warning.html#ruby-27.

### CODE REVIEW

File: cron_swanson.gemspec [hunk 1]
```diff
@@ -23,7 +23,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   # https://github.com/alexdean/cron_swanson/issues/2
```
Comment (alexdean) [inline cron_swanson.gemspec]: can remove this line also ^

**Review by alexdean — COMMENTED** [review]
hi! thanks for this.

please add ruby 3.0 and 3.1 to https://github.com/alexdean/cron_swanson/blob/660bfe24e0b59312a38763087efdbe12e194f399/.github/workflows/rspec.yml#L19, and let's make sure the test suite passes.

**Comment by mzagaja** [comment]
Copy that. Test suite worked fine on 3.1 locally for me, so will see how it looks on CI once it sees the new versions pushed in the new commit.

**Comment by alexdean** [comment]
hm. not sure why tests aren't running on this branch. i'm not super familiar with github actions, so not sure what the deal is here.

edit: maybe it's because the branch is still in your fork of the repo? not sure.

i'm going to merge it, and hopefully the CI will trigger once this lands in master.

**Comment by alexdean** [comment]
looks like all the matrix jobs ran successfully.

https://github.com/alexdean/cron_swanson/actions/runs/2749043998

i'll release a new version to rubygems soon. this will become v0.4.0 i think.

**Comment by mzagaja** [comment]
Thanks @alexdean! I think it is in the case when I forked the repo, it disabled GitHub Actions by default for my fork, and did not run it. So that was a learning experience. Appreciate you getting to this so quickly.

**Comment by alexdean** [comment]
released. https://rubygems.org/gems/cron_swanson/versions/0.4.0

**Pull request merged**